docs/howto/query.md

```diff
@@ -0,0 +1,227 @@
+(query)=
+# How to query the database
+
+## Create/Make your types usable by SQLair
+The first step in building your query is to decide what you would like to put
+into/get out of the database. Identify the types in your Go program that hold
+this information or create them if they don't already exist.
+
+Make sure that the types are in the right format to work with SQLair.
+
+For example:
+```go
+type Employee struct {
+    ID         string `db:"id"`
+    Name       string `db:"name"`
+    LocationID string `db:"location_id"`
+}
+
+type Location struct {
+    ID   string `db:"id"`
+    City string `db:"city"`
+}
+```
+
+> See more: {ref}`types` 
+
+## Write your query string using the types
+
+SQLair queries are regular SQL queries with special input and output expressions
+inserted which allow you to reference the Go types directly from the SQL.
+
+Write your SQL query with SQLair input expressions instead of query parameters
+and SQLair output expressions instead of columns to select.
+
+For example:
+```
+query := `
+    SELECT &Employee.*
+    FROM employee
+    WHERE location_id = $Location.id
+`
+```
+
+> See more {ref}`input-expression-syntax`, {ref}`output-expression-syntax`.
+
+## Prepare your query string to get a Statement
+
+Now you have your query string, you need to pass it to `sqlair.Prepare` along
+with all the types mentioned in the input and output expressions. 
+
+SQLair uses the type information to generate the SQL and verify the input/output
+expressions. If there is an issue with the actual SQL in the query, this will
+not catch it. The returned `Statement` holds the parsed and verified query.
+
+
+For example:
+```go
+stmt, err := sqlair.Prepare(ctx, query, Employee{}, Location{})
+if err != nil {
+    return err
+}
+```
+
+> See more [`sqlair.Preapre`](https://pkg.go.dev/github.com/canonical/sqlair#Prepare)
+
+## Execute the Statement on the database
+
+Execute the statement on a `DB` or a `TX` using the `Query` method on either.
```

Comment:
Here and everywhere else: I'd adopt a formula that is explicit about the goal and the recipe: 

> To execute the statement on a database or a transaction, on your `DB` or `TX` object run the `Query()` method followed by the `Statement` and all the input arguments ... For example... This will return a `Query` object...

Reply:
Sounds good, though I didn't want to put an example here because it should always be run immediately with one of the methods listed below. Putting an example just doing `q := db.Query(ctx, stmt, args...)` might encourage the wrong thing.